File: Hackerrank/ProblemSolving/JumpingOnTheClouds.py
n = int(input())
clouds = list()
for i in range(int(n)):
    x = input()
    clouds.append(int(x))
passed = list()
i = 0
j = 0
while (int(i) < int(n) and int(j) < int(n)):
    i += (int(1), int(2))[i + 2 < n and clouds[int(i)+2] != 1]
    passed.append(int(i))
    j += 1
print((len(passed),len(passed)-1)[passed[len(passed)-1] > n]-1)

File: Hackerrank/ProblemSolving/JumpingonTheClouds.cpp
#include <iostream>
#include <stack>

using namespace std;

int main() {
  int n,j = 0, i = 0;
  cin >> n;
  int clouds[n];
  for(int i = 0;i < n;i++) cin >> clouds[i];
  stack<int> passed;
  while(i < n && j < n) {
    (clouds[i+2] != 1) ? i += 2 : i += 1;
    passed.push(i);
    j++;
  }
  (passed.top() > n) ? cout << passed.size()-1 : cout << passed.size();
  cout << endl;
  return 0;
}
